hornetsnest/include/Static/KTruss/KTruss.cuh

```
@@ -58,35 +101,28 @@ public:
     bool findTrussOfK(bool& stop);
     void runForK(int max_K);
 
-    void runDynamic();
-    bool findTrussOfKDynamic(bool& stop);
-    void runForKDynamic(int max_K);
-
-    void copyOffsetArrayHost(vid_t* host_offset_array);
-    void copyOffsetArrayDevice(vid_t* device_offset_array);
+    void createOffSetArray();
+    void copyOffsetArrayHost(const vert_t* host_offset_array);
+    void copyOffsetArrayDevice(vert_t* device_offset_array);
     void resetEdgeArray();
     void resetVertexArray();
 
-    vid_t getIterationCount();
-    vid_t getMaxK();
+    vert_t getIterationCount();
+    vert_t getMaxK();
+
+    void sortHornet();
 
 private:
     HostDeviceVar<KTrussData> hd_data;
 
-    //load_balancing::BinarySearch load_balancing;
-    //load_balancing::VertexBased1 load_balancing;
+    vert_t originalNE;
+    vert_t originalNV;
 };
 
-//==============================================================================
-
-void callDeviceDifferenceTriangles(const HornetGraph& hornet,
-                                   const gpu::BatchUpdate& batch_update,
-                                   triangle_t* __restrict__ output_triangles,
-                                   int threads_per_intersection,
-                                   int num_intersec_perblock,
-                                   int shifter,
-                                   int thread_blocks,
-                                   int blockdim,
-                                   bool deletion);
+#define CHECK_ERROR(str) \
+    {cudaError_t err; err = cudaGetLastError(); if(err!=0) {printf("ERROR %s:  %d %s\n", str, err, cudaGetErrorString(err)); fflush(stdout); exit(0);}}
```

should throw an exception that python can catch and display to user.  Otherwise it is a log that makes debugging hard

---

Currently not using CHECK_ERROR. Was used for testing and debugging CUDA code. This now removed.